api/client_test.go

```diff
@@ -47,46 +75,347 @@ func Test_DefaultClientConfig_WithEnvVars(t *testing.T) {
+		clientConfig := &ClientConfig{URL: "http://"}
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("missing cert", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.ClientCert = "../testutils/certs/localhost_cert.pem"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("missing key", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.ClientKey = "../testutils/certs/localhost_key.pem"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("invalid cert", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.ClientCert = "foo"
+		clientConfig.TLSConfig.ClientKey = "../testutils/certs/localhost_key.pem"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("invalid key", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.ClientCert = "../testutils/certs/localhost_cert.pem"
+		clientConfig.TLSConfig.ClientKey = "bar"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("invalid CA cert", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.CACert = "foo"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+
+	t.Run("invalid CA path", func(t *testing.T) {
+		clientConfig := BaseClientConfig()
+		clientConfig.TLSConfig.CAPath = "bar"
+
+		c, err := NewClient(clientConfig, nil)
+
+		assert.Nil(t, c)
+		assert.Error(t, err)
+	})
+}
+
+func Test_Client_Status(t *testing.T) {
+	c := &Client{}
+	assert.NotNil(t, c.Status())
+}
+
+func Test_QueryParam_Encode(t *testing.T) {
+	type fields struct {
+		IncludeEvents bool
+		Status        string
+		Run           string
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		{
+			name:   "empty",
+			fields: fields{},
+			want:   "",
+		},
+		{
+			name:   "single field",
+			fields: fields{Status: "foo"},
+			want:   "status=foo",
+		},
+		{
+			name:   "multiple fields",
+			fields: fields{Status: "foo", Run: "bar"},
+			want:   "run=bar&status=foo",
+		},
+		{
+			name:   "multiple fields with include events",
+			fields: fields{Status: "foo", Run: "bar", IncludeEvents: true},
+			want:   "include=events&run=bar&status=foo",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			q := &QueryParam{
+				IncludeEvents: tt.fields.IncludeEvents,
+				Status:        tt.fields.Status,
+				Run:           tt.fields.Run,
+			}
+			assert.Equalf(t, tt.want, q.Encode(), "Encode()")
+		})
+	}
+}
+
+func Test_StatusClient_Overall_BadResponse(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, "foo")
```

Comment:
My IDE marks a linting error here:

```
printf-style function with dynamic format string and no further arguments should use print-style function instead 
```

Updating it to `fmt.Fprint(w, "foo")` (removing the 'f' after "print") resolves it

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/commit/808db068de5e422e1b0ce82a0f8a0c8374509a20